metadata-ingestion/src/datahub/ingestion/source/gc/datahub_gc.py

```diff
@@ -43,34 +55,72 @@ class DataHubGcSourceConfig(ConfigModel):
         description="Sleep between truncation monitoring.",
     )
 
+    dataprocess_cleanup: Optional[DataProcessCleanupConfig] = Field(
+        default=None,
+        description="Configuration for data process cleanup",
+    )
+
+    soft_deleted_entities_cleanup: Optional[SoftDeletedEntitiesCleanupConfig] = Field(
+        default=None,
+        description="Configuration for soft deleted entities cleanup",
+    )
+
 
 @dataclass
-class DataHubGcSourceReport(SourceReport):
+class DataHubGcSourceReport(DataProcessCleanupReport, SoftDeletedEntitiesReport):
     expired_tokens_revoked: int = 0
 
 
 @platform_name("DataHubGc")
 @config_class(DataHubGcSourceConfig)
 @support_status(SupportStatus.TESTING)
 class DataHubGcSource(Source):
+    """
+    DataHubGcSource is responsible for performing garbage collection tasks on DataHub.
+
+    This source performs the following tasks:
+    1. Cleans up expired tokens.
+    2. Truncates Elasticsearch indices based on configuration.
+    3. Cleans up data processes and soft-deleted entities if configured.
+
+    """
+
     def __init__(self, ctx: PipelineContext, config: DataHubGcSourceConfig):
         self.ctx = ctx
         self.config = config
         self.report = DataHubGcSourceReport()
         self.graph = ctx.require_graph("The DataHubGc source")
+        self.dataprocess_cleanup: Optional[DataProcessCleanup] = None
+        self.soft_deleted_entities_cleanup: Optional[SoftDeletedEntitiesCleanup] = None
+
+        if self.config.dataprocess_cleanup:
+            self.dataprocess_cleanup = DataProcessCleanup(
+                ctx, self.config.dataprocess_cleanup, self.report
+            )
+        if self.config.soft_deleted_entities_cleanup:
+            self.soft_deleted_entities_cleanup = SoftDeletedEntitiesCleanup(
+                ctx, self.config.soft_deleted_entities_cleanup, self.report
+            )
 
     @classmethod
     def create(cls, config_dict, ctx):
         config = DataHubGcSourceConfig.parse_obj(config_dict)
         return cls(ctx, config)
 
+    def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
+        return [partial(auto_workunit_reporter, self.get_report())]
```

Comment:
add a comment here - this disables auto status and a couple other things on purpose